Source/Private/ARTGCD.h

```diff
@@ -8,8 +8,34 @@
 
 #import <Foundation/Foundation.h>
 
+/// ARTScheduledBlockHandle wraps a block for delayed invocation within a queue. If/when the handle deallocates, the scheduled invocation will be cancelled if it has not already been executed.
 @interface ARTScheduledBlockHandle : NSObject
+- (instancetype)initWithDelay:(NSTimeInterval)delay queue:(dispatch_queue_t)queue block:(dispatch_block_t)block;
+- (void)cancel;
 @end
 
-ARTScheduledBlockHandle *artDispatchScheduled(NSTimeInterval seconds, dispatch_queue_t queue, dispatch_block_t block);
-void artDispatchCancel(ARTScheduledBlockHandle *handle);
+
+static inline ARTScheduledBlockHandle *artDispatchScheduled(NSTimeInterval seconds, dispatch_queue_t queue, dispatch_block_t block) {
+    // We don't pass the block directly; instead, we put it in a property, and
+    // read it back from the property once the timer fires. This gives us the
+    // chance to set the property to nil when cancelling the timer, thus
+    // releasing our retain on the block early. dispatch_block_cancel doesn't do
+    // this, it retains the block even if you cancel the dispatch until the
+    // dispatch time passes. (How this is a good idea escapes me.)
+    //
+    // From Apple's documentation [1]:
+    //
+    // > Release of any resources associated with the block object is delayed
+    // > until execution of the block object is next attempted (or any execution
+    // > already in progress completes).
+    //
+    // https://developer.apple.com/documentation/dispatch/1431058-dispatch_block_cancel
```

Comment:
I wrote this to preserve the existing behavior / intention, but I'm not sure this is actually needed. 

When using blocks with GCD, they're copied (not retained). Variables referenced outside of the scope of the block have different semantics, but typically the thing we care about is how it affects an objects life cycle and whether or not we can mutate its value from within the block. That said, the resources that "aren't released" are either primitive values (bools, ints, etc) or a _pointer_, all of which have very little overhead.

It seems to me that capturing weak variables is likely the solution to whatever problem this was trying to solve, but I could be wrong!